src/main/scala/temple/generate/server/go/service/GoServiceDAOGenerator.scala

```diff
@@ -1,32 +1,179 @@
 package temple.generate.server.go.service
 
+import temple.ast.{Annotation, Attribute, AttributeType}
+import temple.generate.CRUD
+import temple.generate.CRUD._
+import temple.generate.server.ServiceGenerator.verb
+import temple.generate.server.go.common.GoCommonGenerator.generateGoType
+import temple.generate.server.{CreatedByAttribute, IDAttribute}
 import temple.generate.utils.CodeTerm.{CodeWrap, mkCode}
+import temple.generate.utils.CodeUtils
 import temple.utils.FileUtils
+import temple.utils.StringUtils.doubleQuote
+
+import scala.Option.when
+import scala.collection.immutable.ListMap
 
 object GoServiceDAOGenerator {
 
-  private[go] def generateImports(module: String): String = mkCode(
-    "import",
-    CodeWrap.parens.tabbed(
-      """"database/sql"""",
-      """"fmt"""",
-      "",
-      s""""$module/util"""",
-      "// pq acts as the driver for SQL requests",
-      """_ "github.com/lib/pq"""",
-    ),
-  )
+  private[service] def generateImports(attributes: Map[String, Attribute], module: String): String = {
+    // Check if attributes contains an attribute of type date, time or datetime
+    val containsTime =
+      Set[AttributeType](AttributeType.DateType, AttributeType.TimeType, AttributeType.DateTimeType)
+        .intersect(attributes.values.map(_.attributeType).toSet)
+        .nonEmpty
+
+    mkCode(
+      "import",
+      CodeWrap.parens.tabbed(
+        doubleQuote("database/sql"),
+        //doubleQuote("fmt"),
+        when(containsTime) { doubleQuote("time") },
+        "",
+        //doubleQuote(s"$module/util"),
+        doubleQuote("github.com/google/uuid"),
+        "",
+        "// pq acts as the driver for SQL requests",
+        s"_ ${doubleQuote("github.com/lib/pq")}",
+      ),
+    )
+  }
+
+  private def generateDatastoreInterfaceFunctionReturnType(serviceName: String, operation: CRUD): String =
+    operation match {
+      case ReadAll                => s"(*[]${serviceName.capitalize}, error)"
+      case Create | Read | Update => s"(*${serviceName.capitalize}, error)"
+      case Delete                 => "error"
+    }
+
+  private def generateDAOFunctionName(operation: CRUD, serviceName: String): String =
+    s"${verb(operation)}${serviceName.capitalize}"
+
+  private def generateDatastoreInterfaceFunction(
+    serviceName: String,
+    operation: CRUD,
+    enumByCreatedBy: Boolean,
+  ): String = {
+    val functionName = generateDAOFunctionName(operation, serviceName)
+    mkCode(
+      if (enumByCreatedBy || operation != CRUD.ReadAll) s"$functionName(input ${functionName}Input)"
+      else s"$functionName()",
```

Comment:
I'd be tempted to pull this out to a variable too?

Reply:
I’d pull only the contents of the parens out, so we have `s"$functionName($functionArgs)"` here